### PR TITLE
Added support to display string/enum array as multi select dropdown in AdminUI

### PR DESCRIPTION
**Pull request by siowhoi**
1. Role in APIKey has been updated to enum array
2. Role will be displayed as multi select dropdown in AdminUI
3. Added unit test to assign role to an API key.

Screenshot 2024-02-06 at 11 25 28 am

### CODE REVIEW

File: src/examples/rest/database.sql [hunk 1]
```diff
@@ -44,7 +44,7 @@ CREATE TABLE api_key (
   api_key VARCHAR(255) NOT NULL UNIQUE,
   secret VARCHAR(255) NOT NULL,
   revoked BOOLEAN NOT NULL DEFAULT false,
-  roles JSON NOT NULL DEFAULT (JSON_ARRAY())
+  roles TEXT NOT NULL
```
Comment (tnodell) [inline src/examples/rest/database.sql:47]: Is this change necessary? I would expect our docs and examples to break if we changed what kind of thing roles are. @steve-keep

Comment (steve-keep) [reply src/examples/rest/database.sql:47]: MySQL does not have array support so there are two options TEXT or JSON.

I prefer JSON as you know that it is validated when saved to the DB but it is not perfect.

@siowhoi I assume you had to change this because the JSON type was not compatible with the @Enum decorator?

File: src/examples/rest/src/backend/entities/mysql/api-key.ts [hunk 2]
```diff
@@ -16,6 +17,6 @@ export class ApiKey extends BaseEntity implements ApiKeyStorage {
 	@Property({ type: Boolean, default: false })
 	revoked!: boolean;
 
-	@Property({ type: ArrayType, default: [] })
-	roles!: string[];
+	@Enum({ type: 'enumArray', items: () => Roles, array: true, default: [] })
```
Comment (steve-keep) [inline src/examples/rest/src/backend/entities/mysql/api-key.ts:20]: Do you need to include `array:true` here?

File: src/packages/server/src/metadata-service/resolver.ts [hunk 3]
```diff
@@ -92,13 +92,13 @@ export const getAdminUiMetadataResolver = (hooks?: AdminMetadata['hooks']) => {
...
 						const typeValue = field.getType() as any;
-						const typeName = typeValue.name ? typeValue.name : enumMetadata.get(typeValue)?.name;
+						const typeName = typeValue.name ?? enumMetadata.get(typeValue)?.name;
 
 						const relatedObject = objectTypeData[typeName];
 
 						const fieldObject: AdminUiFieldMetadata = {
 							name: field.name,
-							type: relatedObject?.name || typeName,
+							type: relatedObject?.name || (field.typeOptions.array && `${typeName}[]`) || typeName,
```
Comment (steve-keep) [inline src/packages/server/src/metadata-service/resolver.ts:101]: I thought we decided to send the property `isArray` down to the client since the type did not have the string value `[]`? 

Do you prefer to concat the string like this?

File: src/packages/admin-ui-components/src/detail-panel/component.tsx [hunk 4]
```diff
@@ -71,9 +71,11 @@ const getField = ({ field }: { field: EntityField }) => {
 	}
 
 	const { enumByName } = useSchema();
-	const enumField = enumByName(field.type);
+	const enumField = enumByName(field.type.replace('[]', ''));
 	if (enumField) {
-		return <EnumField name={field.name} typeEnum={enumField} />;
+		return (
+			<EnumField name={field.name} typeEnum={enumField} multiple={field.type.endsWith('[]')} />
```
Comment (steve-keep) [inline src/packages/admin-ui-components/src/detail-panel/component.tsx:77]: Depending on whether we add the `isArray` flag here this might be `field.isArray`

File: src/packages/server/src/metadata-service/resolver.ts (hunk 3 above)
Comment (siowhoi) [reply src/packages/server/src/metadata-service/resolver.ts]: Fixed. Added new property called `isArray`.

File: src/packages/admin-ui-components/src/detail-panel/component.tsx (hunk 4 above)
Comment (siowhoi) [reply src/packages/admin-ui-components/src/detail-panel/component.tsx]: Fixed. Added new property called `isArray`.

File: src/examples/rest/src/backend/entities/mysql/api-key.ts (hunk 2 above)
Comment (siowhoi) [reply src/examples/rest/src/backend/entities/mysql/api-key.ts:20]: Yes, if i remove `array: true`, i get error: `Expected Iterable`
Screenshot 2024-02-07 at 10 40 48 am

File: src/examples/rest/database.sql (hunk 1 above)
Comment (siowhoi) [reply src/examples/rest/database.sql:47]: Yes, I get error: Invalid JSON text. To fix this error, I will need a customType to convert the values to valid JSON string.

File: src/packages/admin-ui-components/src/detail-panel/component.tsx [hunk 5]
```diff
@@ -71,9 +71,9 @@ const getField = ({ field }: { field: EntityField }) => {
 	}
 
 	const { enumByName } = useSchema();
-	const enumField = enumByName(field.type);
+	const enumField = enumByName(field.type.replace('[]', ''));
```
Comment (steve-keep) [inline src/packages/admin-ui-components/src/detail-panel/component.tsx:74]: is this replace still needed?

Comment (siowhoi) [reply src/packages/admin-ui-components/src/detail-panel/component.tsx]: Thanks for pointing it out. I have cleanup the code.
